### ISSUE TITLE
docker-compose up error

### ISSUE THREAD

**Issue by stoltzmaniac** [mikkelkrogsholm/encrypted_dashboard#2]
I make it through your documentation up until: "docker-compose up" upon entering that command I receive:

ERROR: yaml.parser.ParserError: while parsing a block mapping
  in "./docker-compose.yml", line 1, column 1
expected <block end>, but found '<block mapping start>'
  in "./docker-compose.yml", line 89, column 3

I tried this at first by changing all of the domains and usernames as expected, and then simply tried it without changing a single thing - the results were the same.

Any ideas? Thanks.

**Comment by mikkelkrogsholm** [comment]
Did you solve this or is the issue still open?

**Comment by stoltzmaniac** [comment]
No, I could not solve this. I tried it on numerous machines, changing spacing in the yaml file, and a number of other attempts on both digitalocean and google compute engine. Not sure what's going on

**Comment by mikkelkrogsholm** [comment]
Can you provide your compose file?

**Comment by stoltzmaniac** [comment]
I'm using yours: https://github.com/mikkelkrogsholm/encrypted_dashboard/blob/master/docker-compose.yml

**Comment by mikkelkrogsholm** [comment]
Did you change the environment variables?

Den 27. jun. 2017 00.38 skrev "Scott Stoltzman" <notifications@github.com>:

**Comment by stoltzmaniac** [comment]
Exactly, I just did "find/replace" mydomain with stoltzmaniac (and had them setup in GoDaddy)

**Comment by mikkelkrogsholm** [comment]
Can you copy paste the docker compose file? Just leave your password and
user blank

Den 27. jun. 2017 00.41 skrev "Scott Stoltzman" <notifications@github.com>:

**Comment by stoltzmaniac** [comment]
version: "2"

services:
  nginx:
    restart: always
    image: nginx
    container_name: nginx
    ports:
      - "80:80"
      - "443:443"
    volumes:
      - "/etc/nginx/conf.d"
      - "/etc/nginx/vhost.d"
      - "/usr/share/nginx/html"
      - "./volumes/proxy/certs:/etc/nginx/certs:ro"

  nginx-gen:
    restart: always
    image: jwilder/docker-gen
    container_name: nginx-gen
    depends_on:
      - nginx
    volumes:
      - "/var/run/docker.sock:/tmp/docker.sock:ro"
      - "./volumes/proxy/templates/nginx.tmpl:/etc/docker-gen/templates/nginx.tmpl:ro"
    volumes_from:
      - nginx
    entrypoint: /usr/local/bin/docker-gen -notify-sighup nginx -watch -wait 5s:30s /etc/docker-gen/templates/nginx.tmpl /etc/nginx/conf.d/default.conf

  letsencrypt-nginx-proxy-companion:
    restart: always
    image: jrcs/letsencrypt-nginx-proxy-companion
    container_name: letsencrypt-nginx-proxy-companion
    depends_on:
      - nginx
      - nginx-gen
    volumes_from:
      - nginx
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock:ro"
      - "./volumes/proxy/certs:/etc/nginx/certs:rw"
    environment:
      - NGINX_DOCKER_GEN_CONTAINER=nginx-gen

# Create data docker volumes
volumes:
  shiny-apps:
  r-packages:

# Spin up a Shiny docker
shiny:
    restart: always
    image: rocker/shiny
    container_name: shiny
    expose:
      - "3838"
    environment:
      - VIRTUAL_HOST=shiny.stoltzmaniac.com
      - VIRTUAL_NETWORK=nginx-proxy
      - VIRTUAL_PORT=80
      - LETSENCRYPT_HOST=shiny.stoltzmaniac.com
      - LETSENCRYPT_EMAIL=me@myemail.com
    volumes:
      - shiny-apps:/srv/shiny-server/
      - ./volumes/shiny/logs:/var/log/
      - r-packages:/usr/local/lib/R/site-library

# Spin up Rstudio docker

## Use this version to add several users
  # tidyverse:
  #   restart: always
  #   build: ./addusers
  #   container_name: rstudio
  #   depends_on:
  #     - shiny
  #   expose:
  #     - "8787"
  #   environment:
  #     - VIRTUAL_HOST=rstudio.stoltzmaniac.com
  #     - VIRTUAL_NETWORK=nginx-proxy
  #     - VIRTUAL_PORT=80
  #     - LETSENCRYPT_HOST=rstudio.stoltzmaniac.com
  #     - LETSENCRYPT_EMAIL=me@myemail.com
  #   volumes:
  #     - shiny-apps:/home/shiny/apps # Remember to add shiny as a user
  #     - r-packages:/usr/local/lib/R/site-library

  tidyverse:
    restart: always
    image: mikkelkrogsholm/rstudio
    container_name: rstudio
    depends_on:
      - shiny
    expose:
      - "8787"
    environment:
      - VIRTUAL_HOST=rstudio.stoltzmaniac.com
      - VIRTUAL_NETWORK=nginx-proxy
      - VIRTUAL_PORT=80
      - LETSENCRYPT_HOST=rstudio.stoltzmaniac.com
      - LETSENCRYPT_EMAIL=me@myemail.com
      - USER=YOUR_USER
      - PASSWORD=YOUR_PASSWORD
    volumes:
      - shiny-apps:/home/YOUR_USER/apps
      - r-packages:/usr/local/lib/R/site-library

**Comment by stoltzmaniac** [comment]
`
version: "2"

services:
  nginx:
    restart: always
    image: nginx
    container_name: nginx
    ports:
      - "80:80"
      - "443:443"
    volumes:
      - "/etc/nginx/conf.d"
      - "/etc/nginx/vhost.d"
      - "/usr/share/nginx/html"
      - "./volumes/proxy/certs:/etc/nginx/certs:ro"

  nginx-gen:
    restart: always
    image: jwilder/docker-gen
    container_name: nginx-gen
    depends_on:
      - nginx
    volumes:
      - "/var/run/docker.sock:/tmp/docker.sock:ro"
      - "./volumes/proxy/templates/nginx.tmpl:/etc/docker-gen/templates/nginx.tmpl:ro"
    volumes_from:
      - nginx
    entrypoint: /usr/local/bin/docker-gen -notify-sighup nginx -watch -wait 5s:30s /etc/docker-gen/templates/nginx.tmpl /etc/nginx/conf.d/default.conf

  letsencrypt-nginx-proxy-companion:
    restart: always
    image: jrcs/letsencrypt-nginx-proxy-companion
    container_name: letsencrypt-nginx-proxy-companion
    depends_on:
      - nginx
      - nginx-gen
    volumes_from:
      - nginx
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock:ro"
      - "./volumes/proxy/certs:/etc/nginx/certs:rw"
    environment:
      - NGINX_DOCKER_GEN_CONTAINER=nginx-gen

# Create data docker volumes
volumes:
  shiny-apps:
  r-packages:

# Spin up a Shiny docker
shiny:
    restart: always
    image: rocker/shiny
    container_name: shiny
    expose:
      - "3838"
    environment:
      - VIRTUAL_HOST=shiny.stoltzmaniac.com
      - VIRTUAL_NETWORK=nginx-proxy
      - VIRTUAL_PORT=80
      - LETSENCRYPT_HOST=shiny.stoltzmaniac.com
      - LETSENCRYPT_EMAIL=me@myemail.com
    volumes:
      - shiny-apps:/srv/shiny-server/
      - ./volumes/shiny/logs:/var/log/
      - r-packages:/usr/local/lib/R/site-library

# Spin up Rstudio docker

## Use this version to add several users
  # tidyverse:
  #   restart: always
  #   build: ./addusers
  #   container_name: rstudio
  #   depends_on:
  #     - shiny
  #   expose:
  #     - "8787"
  #   environment:
  #     - VIRTUAL_HOST=rstudio.stoltzmaniac.com
  #     - VIRTUAL_NETWORK=nginx-proxy
  #     - VIRTUAL_PORT=80
  #     - LETSENCRYPT_HOST=rstudio.stoltzmaniac.com
  #     - LETSENCRYPT_EMAIL=me@myemail.com
  #   volumes:
  #     - shiny-apps:/home/shiny/apps # Remember to add shiny as a user
  #     - r-packages:/usr/local/lib/R/site-library

  tidyverse:
    restart: always
    image: mikkelkrogsholm/rstudio
    container_name: rstudio
    depends_on:
      - shiny
    expose:
      - "8787"
    environment:
      - VIRTUAL_HOST=rstudio.stoltzmaniac.com
      - VIRTUAL_NETWORK=nginx-proxy
      - VIRTUAL_PORT=80
      - LETSENCRYPT_HOST=rstudio.stoltzmaniac.com
      - LETSENCRYPT_EMAIL=me@myemail.com
      - USER=YOUR_USER
      - PASSWORD=YOUR_PASSWORD
    volumes:
      - shiny-apps:/home/YOUR_USER/apps
      - r-packages:/usr/local/lib/R/site-library



`

**Comment by stoltzmaniac** [comment]
[docker-compose copy.txt](https://github.com/mikkelkrogsholm/encrypted_dashboard/files/1103715/docker-compose.copy.txt)

**Comment by stoltzmaniac** [comment]
I switched to to ".txt" from ".yaml" because it wouldn't upload

**Comment by mikkelkrogsholm** [comment]
Ok. Change shiny image to mikkelkrogsholm/shiny

Run a docker-compose pull

When that's done then run docker-compose up.

Den 27. jun. 2017 00.59 skrev "Scott Stoltzman" <notifications@github.com>:

**Comment by stoltzmaniac** [comment]
Okay, I'll test this out soon, thanks for the help.

**Comment by mikkelkrogsholm** [comment]
Let's see if we can make it work!

Den 27. jun. 2017 01.03 skrev "Scott Stoltzman" <notifications@github.com>:

**Comment by stoltzmaniac** [comment]
Here's what I get:

ERROR: yaml.parser.ParserError: while parsing a block mapping
  in "./docker-compose.yml", line 1, column 1
expected <block end>, but found '<block mapping start>'
  in "./docker-compose.yml", line 89, column 3

**Comment by stoltzmaniac** [comment]
Just theoretically - shouldn't I be able to spin up a virtual machine, install docker, docker-compose, git.

Then just clone your repository and run docker-compose up without actually changing anything? 

**Edit** I mean, just to get past that error :)

**Comment by lstmemery** [comment]
@stoltzmaniac What does your `docker-compose.yml` look like? That message suggests there's an indentation error.

**Comment by stoltzmaniac** [comment]
@lstmemery and @mikkelkrogsholm - for simplicity let's just say say the **exact** file for docker-compose.yml is the one within the original @mikkelkrogsholm repository. Without any tampering with the file at all it should at least get past that error and start to compose. Obviously, the subdomain and whatnot won't work but I would at least like to get past the initial error.

**Comment by lstmemery** [comment]
@stoltzmaniac You are correct. I'll submit a pull request that fixes this shortly.

**Comment by stoltzmaniac** [comment]
Fantastic, thanks @lstmemery I'll check it out when it merges

**Comment by chipoglesby** [comment]
@stoltzmaniac did this ever work?

**Comment by stoltzmaniac** [comment]
I never got it working, but I gave up shortly thereafter and have just been doing things without Docker :( @chipoglesby
